src/container/isReactComponent.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule isReactComponent
+ * @typechecks
+ * @flow
+ */
+
+'use strict';
+
+/**
+ * @internal
+ *
+ * Helper for checking if this is a React Component
+ * created with React.Component or React.createClass().
+ */
+function isReactComponent(component: any): boolean {
```

Comment:
sounds like this is the best way to test for now until React has better support for HOCs setting refs